sql/src/main/java/io/crate/operation/collect/LuceneDocCollector.java

```diff
@@ -245,164 +274,40 @@ private void doCollect(@Nullable OrderedCollectContext collectContext) {
     public void kill() {
         killed = true;
         if (paused.get()) {
-            doCollect(collectContext);
+            downstream.fail(new CancellationException());
+            finishCollect();
         }
     }
 
-
     @Override
     public void pause() {
-        if (orderBy == null) {
-            throw new UnsupportedOperationException();
-        }
         pendingPause = true;
     }
 
     @Override
     public void resume() {
-        if (orderBy == null) {
-            throw new UnsupportedOperationException();
-        }
         if (paused.compareAndSet(true, false)) {
-            doCollect(collectContext);
+            innerCollect();
         }
```

Comment:
can you set pendingPause to false if it is not paused or add a trace/debug logging here?
Had the problem with NL that  if left it paused it will resume right and the right side is then processed in the same thread until it is finishing and resuming left again, but since setNextRow didn't return the pause of the left side didn't get processed.